File: LeetCode_Solutions/LeetCode_Solutions/PowerOfThree_326.m
//
//  PowerOfThree_326.m
//  LeetCode_Solutions
//
//  Created by 来自遥远星系的核心巡洋舰 on 16/11/6.
//  Copyright © 2016年 KnightJoker. All rights reserved.
//




#import "PowerOfThree_326.h"

@implementation PowerOfThree_326


- (BOOL)isPowerOfThree:(NSInteger)n {

    if (n < 0 || n == 0) {
        return NO;
    }
    if ( ((n>>1) & ((n>>1)-1)) == 0) {
        return YES;
    }
    return NO;
}

@end

File: LeetCode_Swift/LeetCode_Swift/PowerOfThree_326.swift
//
//  PowerOfThree_326.swift
//  LeetCode_Swift
//
//  Created by Huni on 18/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

public class PowerOfThree_326 {

    func isPowerOfThree(_ n: Int) -> Bool {
        
        var num = n
        while num > 0 && num % 3 == 0 {
            num /= 3
        }
        return num == 1
    }
}

